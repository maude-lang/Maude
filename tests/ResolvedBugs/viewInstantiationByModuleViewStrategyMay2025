#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude -no-prelude \
  < $srcdir/viewInstantiationByModuleViewStrategyMay2025.maude -no-banner -no-advise \
  > viewInstantiationByModuleViewStrategyMay2025.out 2>&1

diff $srcdir/viewInstantiationByModuleViewStrategyMay2025.expected viewInstantiationByModuleViewStrategyMay2025.out > /dev/null 2>&1
