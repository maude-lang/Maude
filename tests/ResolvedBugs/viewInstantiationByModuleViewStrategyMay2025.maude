***
*** Counterpart of the viewInstantiationByModuleViewApril2025 test for
*** strategy to strategy expression mappings.
***

sth T is
  sort Small .
  strat f : Small @ Small .
endsth

smod M is
  sort Num .
  strat s : Num @ Num .
endsm

view MV from T to M is
  sort Small to Num .
  strat f to s .
endv

sth B is
  sort Tiny .
  strat g : Tiny @ Tiny .
endsth

smod USE-B{X :: B} is
  sd g(N:X$Tiny) := match N:X$Tiny [label "raw thumb"] .
endsm

smod BASE{X :: T} is
  sort Big .
  subsort X$Small < Big .  *** this triggers the bug
endsm

view PV{X :: T} from B to BASE{X} is
  sort Tiny to X$Small .
  strat g(N:Tiny) to expr f(N:X$Small) .
endv

smod TEST is
  pr USE-B{PV{MV}} .
endsm

show sds .
