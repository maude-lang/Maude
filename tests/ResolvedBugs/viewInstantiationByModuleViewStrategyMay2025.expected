sd s(N:Num) := match N:Num .
Bye.
